lib/Dialect/TTNN/Transforms/Passes.cpp

```diff
@@ -233,7 +233,8 @@ class TTNNDecomposeLayouts
     output.tensorMemoryLayout = outputMemoryConfig.getTensorMemoryLayout();
 
     input.shardShape = inputLayoutAttr.getShardShape();
```

Comment:
The problem was in this line, because again `inputLayoutAttr.getShardShape()` is returning `SmallVector`, so by the end of this statement, input.shardShape will have a dangling reference.

Alternatively, this PR https://github.com/tenstorrent/tt-mlir/pull/1566 will also result in fix. @jnie-TT do you know if there is a strong guarantee that `shardShape` in `LayoutInfo` will never be a dangling reference. This fix with `SmallVector` is generally on the safer side, but if there are strong guarantees maybe we can go with `ArrayRef` for a perf reason (though a perf penalty for `SmallVector` here seems rather small).